other-guides/env-files.md

```diff
@@ -0,0 +1,151 @@
+
+  if (command === 'ping') {
+    msg.reply('Pong!');
+  }
+});
+
+// Here you can login the bot with the environment variable you defined above
+client.login(token);
+```
+This is all you will need to get off the ground. Now I will explain why this is important and why you should use environment variables with a `.env` file over a `config.json` file.
+
+## Using Git (ex. GitHub)
+If you're going to publish your code to the public, for example, a Heroku application, then it's imperative you secure information such as your bot token, other API tokens, and any database details. Setting a `.env` file will allow you to do this, but there is an extra step you're going to have to take. And that's using a `.gitignore` file. Simply add the `.env` file to the `.gitignore` file in your local repository. From there on out, any future commits will ignore that file. You can set up a `.env.example` file in its place, but that's not always necessary.
+```
+.env
+```
+To read more information on `.gitignore` files, read [Using Git to share and update code](https://anidiots.guide/other-guides/using-git-to-share-and-update-code#ignoring-files) here on An Idiot's Guide.
+
+If you don't add your `.env` to the `.gitignore` file then you risk your bot token being compromised. Don't let it come to that. The proper security measures.
+
+If you follow this properly, all users will see in your bot code is `process.env.ENV_VARIABLE` which exposes nothing!
+
+## Using Heroku
+Heroku is a website that allows you to start out and host your applications for free. In this case, your Discord bot. However, Heroku requires that you use environment variables. If you setup your files with the `dotenv` package and required the specific environment variables in your code, then all you have to do is go the environment variables in your Heroku application and add the key and value.
+
+In Heroku, these variables are called "Config Vars". I'm not going to go in-depth here about Heroku, but the same way you setup your `.env` file, you would add in new environment variables in Heroku.
+
+![Heroku Config Vars](https://i.imgur.com/MSmEO5K.png)
+
+The only way your bot token can be exposed along with other environment variables is if you do one or more of these things:
+- a) You accidentally expose your `.env` file because you didn't add it to `.gitigonre`
+- b) You added a collaborator through Heroku. A collaborator has almost full permissions as the app owner
+
+Both a and b can be controlled. It's just you need to be smart.
+
+## Set environment variables in the start script
+When starting your application, either locally on your computer via the command line, in a npm start script in your `package.json` or even in a `Dockerfile` you can set what environment your bot should run in. For example, you may want to run your bot in production on Heroku or Glitch and in development on your computer. You can simply do that via adding new scripts in your `package.json`.
```

Comment:
Another running sentence, try breaking it up at `should run in. For example`

Reply:
Will review this (Will resolve the conversation once the update has been made).